drivers/sensor/lsm303dlhc_magn/lsm303dlhc_magn.c

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2018 Philémon Jaermann
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <i2c.h>
+#include <init.h>
+#include <sensor.h>
+#include <logging/sys_log.h>
+
+#include "lsm303dlhc_magn.h"
+
+static int lsm303dlhc_sample_fetch(struct device *dev,
+				   enum sensor_channel chan)
+{
+	struct lsm303dlhc_magn_data *drv_data = dev->driver_data;
+	u8_t tries = 11;
```

Comment:
Why 11?

Reply:
The original author has replied regarding to this in PR #6420:

"Mh I thought about keeping the 10 retries but having the sleep time to be ODR/10, or maybe even 11 retries if the querry is perfectly synchronised with the ODR."

Reply:
We should use a data ready interrupt rather than polling for the flag. Please remove the polling loop for now, you can add the interrupt handling in another PR.